src/ui/handlers/menu-ui-handler.ts

```diff
@@ -337,7 +337,7 @@ export class MenuUiHandler extends OptionSelectUiHandler {
               ui.playSelect();
               const dialogueTestName = sanitizedName;
               const dialogueName = decodeURIComponent(escape(atob(dialogueTestName)));
-              const handler = ui.getHandler() as AwaitableUiHandler;
+              const handler = ui.getCurrentHandler() as AwaitableUiHandler;
```

Comment:
NIT: use generic param `<T>`

```suggestion
              const handler = ui.getCurrentHandler<AwaitableUiHandler>();
```